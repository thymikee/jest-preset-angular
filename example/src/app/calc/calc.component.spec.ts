import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { configureTests, ConfigureFn } from '@lib/testing';

import { CalcComponent } from 'app/calc/calc.component';

describe('CalcComponent', () => {
  let component: CalcComponent;
  let fixture: ComponentFixture<CalcComponent>;

  beforeEach(
    async(() => {
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [CalcComponent]
        });
      };

      configureTests(configure).then(testBed => {
        fixture = testBed.createComponent(CalcComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
      });
    })
  );

  it('should snap', () => {
    expect(fixture).toMatchSnapshot();
  });
});
