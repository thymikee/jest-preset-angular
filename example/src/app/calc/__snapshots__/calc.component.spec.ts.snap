// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalcComponent should snap 1`] = `
<app-calc
  hasAClass="false"
  image={[Function String]}
  prop1={[Function Number]}
>
  <p
    class="a-default-class"
    ng-reflect-klass="a-default-class"
    ng-reflect-ng-class="[object Object]"
  >
     calc works! 1337 another text node test-image-stub 
  </p>
</app-calc>
`;
